tendrl/node_agent/node_sync/disk_sync.py

```diff
@@ -2,212 +2,308 @@
                     devlist["bios_id"] = \
-                        line.split(':')[1].lstrip()
-                elif "Geometry (Logical)" in line:
+                        disk.split(':')[1].lstrip()
+                elif key.strip() == "Geometry (Logical)":
                     devlist["geo_logical"] = \
-                        line.split(':')[1].lstrip()
-                elif "Geometry (BIOS EDD)" in line:
+                        disk.split(':')[1].lstrip()
+                elif key.strip() == "Capacity":
+                    devlist["size"] = \
+                        disk.split('(')[1].split()[0]
+                elif key.strip() == "Geometry (BIOS EDD)":
                     devlist["geo_bios_edd"] = \
-                        line.split(':')[1].lstrip()
-                elif "Geometry (BIOS Legacy)" in line:
+                        disk.split(':')[1].lstrip()
+                elif key.strip() == "Size (BIOS EDD)":
+                    devlist["size_bios_edd"] = \
+                        disk.split(':')[1].lstrip()
+                elif key.strip() == "Geometry (BIOS Legacy)":
                     devlist["geo_bios_legacy"] = \
-                        line.split(':')[1].lstrip()
-            if devlist["disk_type"] == "disk":
-                all_disks.append(devlist)
-            elif devlist["disk_type"] == "partition":
-                devlist["used"] = True
-                disks.append(devlist)
-                parents.append(devlist["parent_id"])
-        for disk in all_disks:
-            if not disk["disk_id"] in parents:
-                disk["used"] = False
-                disks.append(disk)
-    return disks, err
+                        disk.split(':')[1].lstrip()
+                elif key.strip() == "Config Status":
+                    devlist["config_status"] = \
+                        disk.split(':')[1].lstrip()
+            if (devlist["vendor"] != "" and
+                devlist["device"] != "" and
+                devlist["serial_no"] != ""):
+                devlist["disk_id"] = (devlist["vendor"] + "_" +
+                    devlist["device"] + "_" + devlist["serial_no"])
+            elif "virtio" in devlist["driver"]:
+                reg = re.compile(".+\/by-id\/(virtio-.+,).")
+                disk_id = reg.findall(devlist['device_files'])[0]
+                devlist["disk_id"] = disk_id.replace('-', '_').replace(',', '')
+            else:
+                err = "Unable to create disk id for disk: %s" % devlist["disk_name"]
+            if not err:
+                disks[devlist["disk_id"]] = devlist
+                disks_map[devlist['hardware_id']] = devlist["disk_id"]
+            else:
+                Event(
+                    Message(
+                            priority="error",
+                            publisher=NS.publisher_id,
+                            payload={"message": err}
+                        )
+                )
+    return disks, disks_map, err
```

Comment:
raise ValueError if there is any "err" , catch the ValueError in calling function